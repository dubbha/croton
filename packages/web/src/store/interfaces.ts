import { RouterState } from 'connected-react-router';
import { AuthState } from './auth';
import { ShelfState } from './shelf';
import { NotificationsState } from './notification';

export interface AppState {
  router: RouterState;
  auth: AuthState;
  shelf: ShelfState;
  notifications: NotificationsState;
}
