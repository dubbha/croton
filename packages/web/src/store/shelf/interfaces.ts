import { Actions } from 'constants/actions';

export type Shelf = {
  id: number,
  name: string,
  location: string,
  description: string,
  pictureUrl: string,
}

export type Flower = {
  id: number,
  name: string,
  description: string,
  pictureUrls: string[],
  rrules: { [key in Actions]: string },
  shelfId: number,
}

export interface ShelfState {
  isLoading: boolean;
  error: string | null;
  info: string | null;
  shelves: Shelf[];
  flowers: Flower[];
  flower: Flower | null;
}
