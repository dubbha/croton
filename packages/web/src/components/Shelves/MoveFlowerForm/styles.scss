.btn {
  width: 100%;
}

form.moveFlowerForm {
  width: 100%;
}
