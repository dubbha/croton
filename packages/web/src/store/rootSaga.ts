import { all } from 'redux-saga/effects';
import { authLoginSaga, registerSaga } from './auth/sagas';

export function* rootSaga() {
  yield all([
    authLoginSaga(),
    registerSaga(),
  ])
}
