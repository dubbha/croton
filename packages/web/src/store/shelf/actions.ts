import { Actions } from 'constants/actions';
import { Shelf, Flower } from './interfaces';

export const SHELF_INVITE = 'SHELF_INVITE';
export const SHELF_INVITE_SUCCESS = 'SHELF_INVITE_SUCCESS';
export const SHELF_INVITE_ERROR = 'SHELF_INVITE_ERROR';
export const SHELF_INVITE_RESET = 'SHELF_INVITE_RESET';

export const SHELF_INVITE_ACCEPT = 'SHELF_INVITE_ACCEPT';
export const SHELF_INVITE_ACCEPT_SUCCESS = 'SHELF_INVITE_ACCEPT_SUCCESS';
export const SHELF_INVITE_ACCEPT_ERROR = 'SHELF_INVITE_ACCEPT_ERROR';

export const SHELF_DELETE_USER = 'SHELF_DELETE_USER';
export const SHELF_DELETE_USER_SUCCESS = 'SHELF_DELETE_USER_SUCCESS';
export const SHELF_DELETE_USER_ERROR = 'SHELF_DELETE_USER_ERROR';

export const SHELF_ADD_SHELF = 'SHELF_ADD_SHELF';
export const SHELF_ADD_SHELF_SUCCESS = 'SHELF_ADD_SHELF_SUCCESS';
export const SHELF_ADD_SHELF_ERROR = 'SHELF_ADD_SHELF_ERROR';

export const SHELF_EDIT_SHELF = 'SHELF_EDIT_SHELF';
export const SHELF_EDIT_SHELF_SUCCESS = 'SHELF_EDIT_SHELF_SUCCESS';
export const SHELF_EDIT_SHELF_ERROR = 'SHELF_EDIT_SHELF_ERROR';

export const SHELF_DELETE_SHELF = 'SHELF_DELETE_SHELF';
export const SHELF_DELETE_SHELF_SUCCESS = 'SHELF_DELETE_SHELF_SUCCESS';
export const SHELF_DELETE_SHELF_ERROR = 'SHELF_DELETE_SHELF_ERROR';

export const SHELF_GET_SHELVES = 'SHELF_GET_SHELVES';
export const SHELF_GET_SHELVES_SUCCESS = 'SHELF_GET_SHELVES_SUCCESS';
export const SHELF_GET_SHELVES_ERROR = 'SHELF_GET_SHELVES_ERROR';

export const SHELF_ADD_FLOWER = 'SHELF_ADD_FLOWER';
export const SHELF_ADD_FLOWER_SUCCESS = 'SHELF_ADD_FLOWER_SUCCESS';
export const SHELF_ADD_FLOWER_ERROR = 'SHELF_ADD_FLOWER_ERROR';

export const SHELF_EDIT_FLOWER = 'SHELF_EDIT_FLOWER';
export const SHELF_EDIT_FLOWER_SUCCESS = 'SHELF_EDIT_FLOWER_SUCCESS';
export const SHELF_EDIT_FLOWER_ERROR = 'SHELF_EDIT_FLOWER_ERROR';

export const SHELF_DELETE_FLOWER = 'SHELF_DELETE_FLOWER';
export const SHELF_DELETE_FLOWER_SUCCESS = 'SHELF_DELETE_FLOWER_SUCCESS';
export const SHELF_DELETE_FLOWER_ERROR = 'SHELF_DELETE_FLOWER_ERROR';

export const SHELF_GET_FLOWERS = 'SHELF_GET_FLOWERS';
export const SHELF_GET_FLOWERS_SUCCESS = 'SHELF_GET_FLOWERS_SUCCESS';
export const SHELF_GET_FLOWERS_ERROR = 'SHELF_GET_FLOWERS_ERROR';

export const SHELF_GET_FLOWER = 'SHELF_GET_FLOWER';
export const SHELF_GET_FLOWER_SUCCESS = 'SHELF_GET_FLOWER_SUCCESS';
export const SHELF_GET_FLOWER_ERROR = 'SHELF_GET_FLOWER_ERROR';

export const SHELF_ACTION = 'SHELF_ACTION';
export const SHELF_ACTION_SUCCESS = 'SHELF_ACTION_SUCCESS';
export const SHELF_ACTION_ERROR = 'SHELF_ACTION_ERROR';

export const SHELF_GET_LAST_ACTIONS = 'SHELF_GET_LAST_ACTIONS';
export const SHELF_GET_LAST_ACTIONS_SUCCESS = 'SHELF_GET_LAST_ACTIONS_SUCCESS';
export const SHELF_GET_LAST_ACTIONS_ERROR = 'SHELF_GET_LAST_ACTIONS_ERROR';

export const SHELF_RESET = 'SHELF_RESET';

export interface ShelfInvite {
  type: typeof SHELF_INVITE;
  payload: { userEmail: string; shelfId: number };
}

export interface ShelfInviteSuccess {
  type: typeof SHELF_INVITE_SUCCESS;
  payload: { info: string };
}

export interface ShelfInviteError {
  type: typeof SHELF_INVITE_ERROR;
  payload: { error: string };
}

export interface ShelfInviteAccept {
  type: typeof SHELF_INVITE_ACCEPT;
  payload: { shelfInvitationToken: string };
}

export interface ShelfInviteAcceptSuccess {
  type: typeof SHELF_INVITE_ACCEPT_SUCCESS;
  payload: { info: string };
}

export interface ShelfInviteAcceptError {
  type: typeof SHELF_INVITE_ACCEPT_ERROR;
  payload: { error: string };
}

export interface ShelfDeleteUser {
  type: typeof SHELF_DELETE_USER;
  payload: { shelfId: number, userId: number };
}

export interface ShelfDeleteUserSuccess {
  type: typeof SHELF_DELETE_USER_SUCCESS;
  payload: { info: string };
}

export interface ShelfDeleteUserError {
  type: typeof SHELF_DELETE_USER_ERROR;
  payload: { error: string };
}

export interface ShelfAddShelf {
  type: typeof SHELF_ADD_SHELF;
  payload: { name: string, description: string, location: string };
}

export interface ShelfAddShelfSuccess {
  type: typeof SHELF_ADD_SHELF_SUCCESS;
  payload: { info: string };
}

export interface ShelfAddShelfError {
  type: typeof SHELF_ADD_SHELF_ERROR;
  payload: { error: string };
}

export interface ShelfEditShelf {
  type: typeof SHELF_EDIT_SHELF;
  payload: { id: number, name: string, description: string, location: string };
}

export interface ShelfEditShelfSuccess {
  type: typeof SHELF_EDIT_SHELF_SUCCESS;
  payload: { info: string };
}

export interface ShelfEditShelfError {
  type: typeof SHELF_EDIT_SHELF_ERROR;
  payload: { error: string };
}

export interface ShelfDeleteShelf {
  type: typeof SHELF_DELETE_SHELF;
  payload: { id: number };
}

export interface ShelfDeleteShelfSuccess {
  type: typeof SHELF_DELETE_SHELF_SUCCESS;
  payload: { info: string };
}

export interface ShelfDeleteShelfError {
  type: typeof SHELF_DELETE_SHELF_ERROR;
  payload: { error: string };
}

export interface ShelfGetShelves {
  type: typeof SHELF_GET_SHELVES;
}

export interface ShelfGetShelvesSuccess {
  type: typeof SHELF_GET_SHELVES_SUCCESS;
  payload: { shelves: Shelf[] };
}

export interface ShelfGetShelvesError {
  type: typeof SHELF_GET_SHELVES_ERROR;
  payload: { error: string };
}

export interface ShelfAddFlower {
  type: typeof SHELF_ADD_FLOWER;
  payload: {
    shelfId: number,
    name: string,
    description: string,
    rrules: { [key in Actions]?: string }
  };
}

export interface ShelfAddFlowerSuccess {
  type: typeof SHELF_ADD_FLOWER_SUCCESS;
  payload: { info: string };
}

export interface ShelfAddFlowerError {
  type: typeof SHELF_ADD_FLOWER_ERROR;
  payload: { error: string };
}

export interface ShelfEditFlower {
  type: typeof SHELF_EDIT_FLOWER;
  payload: {
    id: number,
    shelfId: number,
    name: string,
    description: string,
    rrules: { [key in Actions]?: string },
  };
}

export interface ShelfEditFlowerSuccess {
  type: typeof SHELF_EDIT_FLOWER_SUCCESS;
  payload: { info: string };
}

export interface ShelfEditFlowerError {
  type: typeof SHELF_EDIT_FLOWER_ERROR;
  payload: { error: string };
}

export interface ShelfDeleteFlower {
  type: typeof SHELF_DELETE_FLOWER;
  payload: { id: number, shelfId: number };
}

export interface ShelfDeleteFlowerSuccess {
  type: typeof SHELF_DELETE_FLOWER_SUCCESS;
  payload: { info: string };
}

export interface ShelfDeleteFlowerError {
  type: typeof SHELF_DELETE_FLOWER_ERROR;
  payload: { error: string };
}

export interface ShelfGetFlowers {
  type: typeof SHELF_GET_FLOWERS;
  payload: { shelfId: number };
}

export interface ShelfGetFlowersSuccess {
  type: typeof SHELF_GET_FLOWERS_SUCCESS;
  payload: { flowers: Flower[] };
}

export interface ShelfGetFlowersError {
  type: typeof SHELF_GET_FLOWERS_ERROR;
  payload: { error: string };
}

export interface ShelfGetFlower {
  type: typeof SHELF_GET_FLOWER;
  payload: { id: number };
}

export interface ShelfGetFlowerSuccess {
  type: typeof SHELF_GET_FLOWER_SUCCESS;
  payload: { flower: Flower };
}

export interface ShelfGetFlowerError {
  type: typeof SHELF_GET_FLOWER_ERROR;
  payload: { error: string };
}

export interface ShelfAction {
  type: typeof SHELF_ACTION;
  payload: { action: Actions, flowerId: number, shelfId: number };
}

export interface ShelfActionSuccess {
  type: typeof SHELF_ACTION_SUCCESS;
  payload: { info: string };
}

export interface ShelfActionError {
  type: typeof SHELF_ACTION_ERROR;
  payload: { error: string };
}

export interface ShelfGetLastActions {
  type: typeof SHELF_GET_LAST_ACTIONS;
  payload: { flowerId: number, shelfId: number };
}

export interface ShelfGetLastActionsSuccess {
  type: typeof SHELF_GET_LAST_ACTIONS_SUCCESS;
  payload: {
    [key in Actions]: {
      timestamp: number;
      user: { id: number, firstName: string, lastName: string };
    }
  }
}

export interface ShelfGetLastActionsError {
  type: typeof SHELF_GET_LAST_ACTIONS_ERROR;
  payload: { error: string };
}

export interface ShelfReset {
  type: typeof SHELF_RESET;
}

export type ShelfActionTypes =
  | ShelfInvite
  | ShelfInviteSuccess
  | ShelfInviteError
  | ShelfInviteAccept
  | ShelfInviteAcceptSuccess
  | ShelfInviteAcceptError
  | ShelfDeleteUser
  | ShelfDeleteUserSuccess
  | ShelfDeleteUserError
  | ShelfAddShelf
  | ShelfAddShelfSuccess
  | ShelfAddShelfError
  | ShelfEditShelf
  | ShelfEditShelfSuccess
  | ShelfEditShelfError
  | ShelfDeleteShelf
  | ShelfDeleteShelfSuccess
  | ShelfDeleteShelfError
  | ShelfGetShelves
  | ShelfGetShelvesSuccess
  | ShelfGetShelvesError
  | ShelfAddFlower
  | ShelfAddFlowerSuccess
  | ShelfAddFlowerError
  | ShelfEditFlower
  | ShelfEditFlowerSuccess
  | ShelfEditFlowerError
  | ShelfDeleteFlower
  | ShelfDeleteFlowerSuccess
  | ShelfDeleteFlowerError
  | ShelfGetFlowers
  | ShelfGetFlowersSuccess
  | ShelfGetFlowersError
  | ShelfGetFlower
  | ShelfGetFlowerSuccess
  | ShelfGetFlowerError
  | ShelfAction
  | ShelfActionSuccess
  | ShelfActionError
  | ShelfGetLastActions
  | ShelfGetLastActionsSuccess
  | ShelfGetLastActionsError
  | ShelfReset;
