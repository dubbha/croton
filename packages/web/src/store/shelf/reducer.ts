import { LOCATION_CHANGE, LocationChangeAction } from 'connected-react-router';
import {
  SHELF_INVITE,
  SHELF_INVITE_SUCCESS,
  SHELF_INVITE_ERROR,
  SHELF_INVITE_ACCEPT,
  SHELF_INVITE_ACCEPT_SUCCESS,
  SHELF_INVITE_ACCEPT_ERROR,
  SHELF_DELETE_USER,
  SHELF_DELETE_USER_SUCCESS,
  SHELF_DELETE_USER_ERROR,
  SHELF_ADD_SHELF,
  SHELF_ADD_SHELF_SUCCESS,
  SHELF_ADD_SHELF_ERROR,
  SHELF_EDIT_SHELF,
  SHELF_EDIT_SHELF_SUCCESS,
  SHELF_EDIT_SHELF_ERROR,
  SHELF_DELETE_SHELF,
  SHELF_DELETE_SHELF_SUCCESS,
  SHELF_DELETE_SHELF_ERROR,
  SHELF_GET_SHELVES,
  SHELF_GET_SHELVES_SUCCESS,
  SHELF_GET_SHELVES_ERROR,
  SHELF_ADD_FLOWER,
  SHELF_ADD_FLOWER_SUCCESS,
  SHELF_ADD_FLOWER_ERROR,
  SHELF_EDIT_FLOWER,
  SHELF_EDIT_FLOWER_SUCCESS,
  SHELF_EDIT_FLOWER_ERROR,
  SHELF_DELETE_FLOWER,
  SHELF_DELETE_FLOWER_SUCCESS,
  SHELF_DELETE_FLOWER_ERROR,
  SHELF_GET_FLOWERS,
  SHELF_GET_FLOWERS_SUCCESS,
  SHELF_GET_FLOWERS_ERROR,
  SHELF_GET_FLOWER,
  SHELF_GET_FLOWER_SUCCESS,
  SHELF_GET_FLOWER_ERROR,
  SHELF_ACTION,
  SHELF_ACTION_SUCCESS,
  SHELF_ACTION_ERROR,
  SHELF_GET_LAST_ACTIONS,
  SHELF_GET_LAST_ACTIONS_SUCCESS,
  SHELF_GET_LAST_ACTIONS_ERROR,
  SHELF_RESET,
  ShelfActionTypes,
} from './actions';
import { ShelfState, Flower } from './interfaces';

export const initialState: ShelfState = {
  isLoading: false,
  error: null,
  info: null,
  shelves: [],
  flowers: [],
  flower: null,
};

export function shelfReducer(
  state = initialState,
  action: ShelfActionTypes | LocationChangeAction,
): ShelfState {
  switch (action.type) {
    case SHELF_INVITE:
    case SHELF_INVITE_ACCEPT:
    case SHELF_DELETE_USER:
    case SHELF_ADD_SHELF:
    case SHELF_EDIT_SHELF:
    case SHELF_DELETE_SHELF:
    case SHELF_GET_SHELVES:
    case SHELF_ADD_FLOWER:
    case SHELF_EDIT_FLOWER:
    case SHELF_DELETE_FLOWER:
    case SHELF_GET_FLOWERS:
    case SHELF_GET_FLOWER:
    case SHELF_ACTION:
    case SHELF_GET_LAST_ACTIONS:
      return {
        ...state,
        isLoading: true,
        error: null,
        info: null,
      };
    case SHELF_INVITE_SUCCESS:
    case SHELF_INVITE_ACCEPT_SUCCESS:
    case SHELF_DELETE_USER_SUCCESS:
    case SHELF_ADD_SHELF_SUCCESS:
    case SHELF_EDIT_SHELF_SUCCESS:
    case SHELF_DELETE_SHELF_SUCCESS:
    case SHELF_ADD_FLOWER_SUCCESS:
    case SHELF_EDIT_FLOWER_SUCCESS:
    case SHELF_DELETE_FLOWER_SUCCESS:
    case SHELF_ACTION_SUCCESS:
      return {
        ...state,
        isLoading: false,
        info: action.payload.info,
      };
    case SHELF_GET_SHELVES_SUCCESS:
      return {
        ...state,
        isLoading: false,
        shelves: action.payload.shelves,
      };
    case SHELF_GET_FLOWERS_SUCCESS:
      return {
        ...state,
        isLoading: false,
        flowers: action.payload.flowers,
      };
    case SHELF_GET_FLOWER_SUCCESS:
      return {
        ...state,
        isLoading: false,
        flower: action.payload.flower,
      };
    case SHELF_GET_LAST_ACTIONS_SUCCESS:
      return {
        ...state,
        isLoading: false,
        flower: {
          ...state.flower as Flower,
          lastActions: action.payload,
        },
      };
    case SHELF_INVITE_ERROR:
    case SHELF_INVITE_ACCEPT_ERROR:
    case SHELF_DELETE_USER_ERROR:
    case SHELF_ADD_SHELF_ERROR:
    case SHELF_EDIT_SHELF_ERROR:
    case SHELF_DELETE_SHELF_ERROR:
    case SHELF_GET_SHELVES_ERROR:
    case SHELF_ADD_FLOWER_ERROR:
    case SHELF_EDIT_FLOWER_ERROR:
    case SHELF_DELETE_FLOWER_ERROR:
    case SHELF_GET_FLOWERS_ERROR:
    case SHELF_GET_FLOWER_ERROR:
    case SHELF_ACTION_ERROR:
    case SHELF_GET_LAST_ACTIONS_ERROR:
      return {
        ...state,
        isLoading: false,
        error: action.payload.error,
        info: null,
      };
    case SHELF_RESET:
    case LOCATION_CHANGE:
      return {
        ...state,
        isLoading: false,
        error: null,
        info: null,
        flower: null,
      };
    default:
      return state;
  }
}
