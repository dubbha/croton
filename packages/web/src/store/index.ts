import { combineReducers, createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { authReducer } from './auth';
import { shelfReducer } from './shelf';
import { rootSaga } from './rootSaga';

export const history = createBrowserHistory();
const sagaMiddleware = createSagaMiddleware();

const authPersistConfig = {
  key: 'auth',
  storage,
  blacklist: ['isLoading', 'info', 'error'],
};

const store = createStore(
  combineReducers({
    router: connectRouter(history),
    auth: persistReducer(authPersistConfig, authReducer),
    shelf: shelfReducer,
  }),
  composeWithDevTools(
    applyMiddleware(routerMiddleware(history), sagaMiddleware),
  ),
);

export default store;

export const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);
