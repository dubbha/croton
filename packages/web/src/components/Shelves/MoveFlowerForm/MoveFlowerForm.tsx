import React, { useState, FormEvent } from 'react';

import { Shelf } from 'store/shelf';

import {
  Form,
  SubmitButton,
  ErrorAlert,
  AlertPlaceholder,
  InfoAlert,
} from 'elements';

import './styles.scss';

type Props = {
  shelves: Shelf[];
  onSubmit: (shelfId: number) => void;
  error: string | null;
  info: string | null;
  flowerName: string | null;
};

export const MoveFlowerForm = ({
  shelves,
  onSubmit,
  error,
  info,
  flowerName,
}: Props) => {
  const [selectedShelfId, setSelectedShelfId] = useState<Shelf['id']>(
    shelves[0].id,
  );

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    onSubmit(selectedShelfId);
  };

  return (
    <Form onSubmit={handleSubmit} className="moveFlowerForm">
      {error ? <ErrorAlert>{error}</ErrorAlert> : <AlertPlaceholder />}
      {info ? <InfoAlert>{info}</InfoAlert> : <AlertPlaceholder />}
      <Form.Group>
        <Form.Label>Choose a new shelf for {flowerName}</Form.Label>
        <Form.Control
          as="select"
          data-testid="moveFlowerForm__selectShelf"
          onChange={(e) => {
            setSelectedShelfId(Number(e.target.value));
          }}
          custom
        >
          {shelves.map((shelf) => (
            <option key={`${shelf.name}-${shelf.description}`} value={shelf.id}>
              {shelf.name}
            </option>
          ))}
        </Form.Control>
      </Form.Group>
      <SubmitButton>
        <span>Move Flower</span>
      </SubmitButton>
    </Form>
  );
};
