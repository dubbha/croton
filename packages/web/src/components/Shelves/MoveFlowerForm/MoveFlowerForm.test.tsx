import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Shelf } from 'store/shelf';
import { MoveFlowerForm } from './MoveFlowerForm';

describe('components/MoveFlowerForm', () => {
  const fn = jest.fn();
  const shelves = [
    { id: 1, name: 'Test Shelf 1' },
    { id: 2, name: 'Test Shelf 2' },
    { id: 3, name: 'Test Shelf 3' },
  ] as Shelf[];
  const info = 'Some info';
  const flowerName = 'Rose';

  it('should render successfully', () => {
    const { container } = render(
      <MoveFlowerForm
        onSubmit={fn}
        shelves={shelves}
        error={null}
        info={null}
        flowerName={flowerName}
      />,
      { wrapper: MemoryRouter },
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onSubmit prop on submit', () => {
    const { container, getByTestId } = render(
      <MoveFlowerForm
        onSubmit={fn}
        shelves={shelves}
        error={null}
        info={null}
        flowerName={flowerName}
      />,
      { wrapper: MemoryRouter },
    );
    expect(container.firstChild).toMatchSnapshot();

    const submitButton = getByTestId('submitButton');
    fireEvent.click(submitButton);
    expect(fn).toBeCalled();
  });

  it('should display error alert on error', () => {
    const error = 'ERROR';
    const { getByTestId } = render(
      <MoveFlowerForm
        onSubmit={fn}
        shelves={shelves}
        error={error}
        info={null}
        flowerName={flowerName}
      />,
      { wrapper: MemoryRouter },
    );

    const errorAlert = getByTestId('errorAlert');

    expect(errorAlert.textContent).toBe('ERROR');
  });

  it('should display info when flower is moved', () => {
    const { getByTestId } = render(
      <MoveFlowerForm
        onSubmit={fn}
        shelves={shelves}
        error={null}
        info={info}
        flowerName={flowerName}
      />,
      { wrapper: MemoryRouter },
    );

    const infoAlert = getByTestId('infoAlert');

    expect(infoAlert.textContent).toBe(info);
  });
});
