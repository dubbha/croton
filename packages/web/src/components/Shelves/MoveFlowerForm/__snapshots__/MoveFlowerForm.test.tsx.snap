// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MoveFlowerForm should call onSubmit prop on submit 1`] = `
<form
  class="moveFlowerForm"
>
  <div
    data-testid="alertPlaceholder"
    style="height: 38px;"
  />
  <div
    data-testid="alertPlaceholder"
    style="height: 38px;"
  />
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      Choose a new shelf for 
      Rose
    </label>
    <select
      class="custom-select"
      data-testid="moveFlowerForm__selectShelf"
    >
      <option
        value="1"
      >
        Test Shelf 1
      </option>
      <option
        value="2"
      >
        Test Shelf 2
      </option>
      <option
        value="3"
      >
        Test Shelf 3
      </option>
    </select>
  </div>
  <button
    class="btn btn-primary"
    data-testid="submitButton"
    type="submit"
  >
    <span>
      Move Flower
    </span>
  </button>
</form>
`;

exports[`components/MoveFlowerForm should render successfully 1`] = `
<form
  class="moveFlowerForm"
>
  <div
    data-testid="alertPlaceholder"
    style="height: 38px;"
  />
  <div
    data-testid="alertPlaceholder"
    style="height: 38px;"
  />
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      Choose a new shelf for 
      Rose
    </label>
    <select
      class="custom-select"
      data-testid="moveFlowerForm__selectShelf"
    >
      <option
        value="1"
      >
        Test Shelf 1
      </option>
      <option
        value="2"
      >
        Test Shelf 2
      </option>
      <option
        value="3"
      >
        Test Shelf 3
      </option>
    </select>
  </div>
  <button
    class="btn btn-primary"
    data-testid="submitButton"
    type="submit"
  >
    <span>
      Move Flower
    </span>
  </button>
</form>
`;
